app/server/utils/ussd/kenya_ussd_state_machine.py

```diff
@@ -112,7 +112,7 @@ def is_valid_pin(self, user_input):
     def new_pins_match(self, user_input):
         pins_match = False
         # get previous pin input
-        initial_pin = self.session.user_input.split('*')[-2]
```

Comment:
user_input to session lags one behind, so it was actually "1234" in the input, not "1234*1234". but saving it to session_data seems safer than depending on the previous input

Reply:
yeah 100 agreed.